### PR TITLE
fix support of mission manipulation commands and provide support for missions coming from a vehicle

### PR DESCRIPTION
**Pull request by TShapinsky**

### CODE REVIEW

**Comment by TShapinsky** [comment]
@ne0fhyk the fixes here relate to auto-mission assistant. Don't worry this doesn't include the crappy hack.

File: ServiceApp/src/org/droidplanner/core/mission/Mission.java [hunk 1]
```diff
@@ -265,9 +269,13 @@ public void onMissionLoaded(List<msg_mission_item> msgs) {
     }
 
     private List<MissionItem> processMavLinkMessages(List<msg_mission_item> msgs) {
+        waypointToMissionItem.clear();
```
Comment (ne0fhyk) [inline ServiceApp/src/org/droidplanner/core/mission/Mission.java]: @TShapinsky if a mission is pushed to the copter when it's unarmed, this method will be called when it's armed, causing the previous set of mission info to be cleared, and replaced. 

In most case that's fine, but complex mission items (survey, structure scan...) will be replaced with the set of simple mission items they are composed of, causing a loss of info about the complex mission items.

Comment (TShapinsky) [inline ServiceApp/src/org/droidplanner/core/mission/Mission.java]: @ne0fhyk I could dump the data from into into temporary variables and them compare to see if I should overwrite what's already on tower. Unless the missions each have some UID (which I doubt) which can be used to identify them.

**Comment by ne0fhyk** [comment]
- [ ] Update the `Mission` drone attribute so that it only contains `SpatialItem` and `Command` mission items.
- [ ] Remove the mission item to waypoint sparse arrays.
- [ ] Remove the mission item from the broadcast.

**Pull request merged**